_commands/basics/cd.md

```diff
@@ -16,7 +16,88 @@ $ cd /home
 
 ### Useful Options / Examples
 
+
 #### `cd -`
 
-Jump back directory
+Jump to the last directory that you were in. This will not always necessarily be the parent to your current directory, which would instead be the case if you were to use `cd ..`
+
+(Note that this command prints current path after completion)
+
+~~~ bash
+Alborz-Brals-MacBook-Pro:basics abral$ pwd
+/Users/abral/Desktop/Computer_Science/EECS 398 (C4CS)/adv_HW_7_2/c4cs.github.io/_commands/basics
+Alborz-Brals-MacBook-Pro:basics abral$ cd /tmp
+Alborz-Brals-MacBook-Pro:tmp abral$ pwd
+/tmp
+Alborz-Brals-MacBook-Pro:tmp abral$ cd -
+/Users/abral/Desktop/Computer_Science/EECS 398 (C4CS)/adv_HW_7_2/c4cs.github.io/_commands/basics
+~~~
+
+
+#### `cd ..`
+
+Another way to jump back directory
+
+The two dot characters `..` represent the parent directory to the current directory in which we are working
+
+~~~ bash
+Alborz-Brals-MacBook-Pro:basics abral$ pwd
+/Users/abral/Desktop/Computer_Science/EECS 398 (C4CS)/adv_HW_7/c4cs.github.io/_commands/basics
+Alborz-Brals-MacBook-Pro:basics abral$ cd ..
+Alborz-Brals-MacBook-Pro:_commands abral$ pwd
+/Users/abral/Desktop/Computer_Science/EECS 398 (C4CS)/adv_HW_7/c4cs.github.io/_commands
+~~~
+
+#### `cd .`
+
+The one dot character `.` represents the current directory in which we are working. This command simply moves us to our current directory
+
+~~~ bash
+Alborz-Brals-MacBook-Pro:basics abral$ pwd
+/Users/abral/Desktop/Computer_Science/EECS 398 (C4CS)/adv_HW_7/c4cs.github.io/_commands/basics
+Alborz-Brals-MacBook-Pro:basics abral$ cd .
+Alborz-Brals-MacBook-Pro:basics abral$ pwd
+/Users/abral/Desktop/Computer_Science/EECS 398 (C4CS)/adv_HW_7/c4cs.github.io/_commands/basics
+~~~
+
+Of course, this accomplishes nothing; however, there are other uses for the `.` character pertaining to the `cd` command:
 
+`.` can mean "source", as in the following example
+
+~~~ bash
+./my_script
+~~~ 
+
+However, note that when we use `.`, it runs "my_script" as an executable, and launches a new shell to do so. On the other hand, if we were to use `source` instead, like so:
```

Comment:
Do you think this introduces more confusion than it helps? The dot can mean lots of things, and if you're going to include this it'd be good to disambiguate a little better:

__As a POSIX shell construct__
Synonymous with `source`, as you mentioned. Used to execute commands in the current shell.

__As a filesystem construct__
`.` and `..` are special filesystem constructs that are [hard links](https://en.wikipedia.org/wiki/Hard_link) to the current and parent directories, respectively.

__As a "dotfile"__
There's a convention that filenames beginning with `.` are hidden by default. (e.g. `.git`)

---

__So__, something like this is perfectly valid:

```
$ . ./.my_script.sh
```

1. Our filename is called `.my_script.sh`.
2. The reason we have to run `./.my_script.sh` is because `.`, or the current working directory, is not in `PATH`. This is _exactly_ the same as the full path, e.g. `/Users/mterwilliger/Desktop/.my_script.sh`, and has nothing to do with the shell. The reason our shell does not search our current directory for executable is security related: we don't want to automatically execute arbitrary programs that happen to be in the current (potentially untrusted) directory.
3. We then want to execute this inline, instead of in a subshell.